win: fix uv_available_parallelism on win32 (#4525)

giordano opened the pull request:
Follow up to #38, backports https://github.com/libuv/libuv/pull/4525

tkelman commented:
This has now passed Julia's `make testall` locally 12 times in a row without freezing. I'm inclined to merge.

yuyichao commented:
:tada:  How often does it freeze before?

tkelman commented:
More often than that. Usually about 4-6 times in a row could get it to happen.

vtjnash commented:
Lgtm

bnoordhuis committed:
win: fix uv_available_parallelism on win32 (#4525)

Fixes commit 58dfb6c89b from a few days ago. DWORD_PTR is 32 bits on
x86 Windows. Use the right bit count when checking the population count.

Interestingly enough, it manifested itself as double counting online
processors, presumably because the compiler emits a ROR instead of SHR.

Fixes: https://github.com/libuv/libuv/issues/4524
(cherry picked from commit 5ff1fc724f7f53d921599dbe18e6f96b298233f1)
